Various visionOS fixes

madsmtm opened the pull request:
A few small mistakes was introduced in https://github.com/rust-lang/rust/pull/121419, probably after the rename from `xros` to `visionos`. See the commits for details.

CC @agg23

Since you reviewed https://github.com/rust-lang/rust/pull/121419
r? davidtwco

madsmtm committed:
visionOS: Fix unused import warning

The import is used once in this file, inside `posix_spawn`, so let's move the import into that function instead, to reduce the number of `cfg`s that need to be kept in sync.

madsmtm committed:
visionOS: Fix logic for finding the SDK root

The `sdk_name` is `xros`/`xrsimulator`, not `visionos`/`visionossimulator`.

agg23 commented:
Yep that's the rename, sorry.

agg23 commented on library/std/src/sys/pal/unix/process/process_unix.rs:
```diff
@@ -466,6 +454,7 @@ impl Command {
         envp: Option<&CStringArray>,
     ) -> io::Result<Option<Process>> {
         use crate::mem::MaybeUninit;
+        use crate::sys::weak::weak;
```
What is this change for and how is it related?

madsmtm replied:
`./x build library --target=aarch64-apple-visions` throws a warning (that gets converted to an error) because the imported `weak` unused - see the commit that introduces it for details.

madsmtm committed:
Allow specifying SDKROOT as containing XRSimulator.platform

Checking this was missing from the `link_env_remove` function, so compilation might fail if set when compiling for macOS

davidtwco commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 1a7238407c8964fb902f259d712eac9e87052680 has been approved by `davidtwco`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=davidtwco 1a7238407c8964fb902f259d712eac9e87052680 -->
<!-- homu: {"type":"Approved","sha":"1a7238407c8964fb902f259d712eac9e87052680","approver":"davidtwco","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
